#!/usr/bin/env python

PACKAGE = "singaboat_vrx"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kP", double_t, 0, "Proportional gain of PID controller", 8.0, 0, 10.0)
gen.add("kI", double_t, 0, "Integral gain of PID controller", 0.01, 0, 0.1)
gen.add("kD", double_t, 0, "Derivative gain of PID controller", 1.1, 0, 5.0)
gen.add("kS", double_t, 0, "Saturation constant of PID controller", 10, 0, 50.0)
gen.add("goal_tol", double_t, 0, "Goal tolerance dead-band of PID controller", 2.5, 0, 10.0)
gen.add("v_const", double_t, 0, "Proportional gain for linear velocity outside goal tolerance", 5.7, 0, 10.0)
gen.add("v_limit", double_t, 0, "Saturation limit for linear velocity outside goal tolerance", 1.0, 0, 2.0)

exit(gen.generate(PACKAGE, "singaboat_vrx", "StationKeeping"))
